Sources/ARKit-CoreLocation/Nodes/PolylineNode.swift

```diff
@@ -82,17 +83,24 @@ private extension PolylineNode {
 
             let bearing = -currentLocation.bearing(between: nextLocation)
 
-            // This pivot translation in the +Z direction will cause the box to be
-            // rendered a bit away from its true location. The actual factor should
-            // probably be a parameter or property of some sort.
-            boxNode.pivot = SCNMatrix4MakeTranslation(0, 0, 0.5 * Float(distance))
+            // Orient the line to point from currentLoction to nextLocation
             boxNode.eulerAngles.y = Float(bearing).degreesToRadians
 
-            let locationNode = LocationNode(location: currentLocation, tag: tag)
+            let locationNode = LocationNode(location: midLoction, tag: tag)
             locationNode.addChildNode(boxNode)
 
             locationNodes.append(locationNode)
         }
     }
 
+    func midPoint (_ first: CLLocation, _ second: CLLocation) -> CLLocation {
```

Comment:
This function uses an incorrect formula. Between two lat/lon positions, we use a GC formula elsewhere in the library. https://edwilliams.org/avform.htm#Intermediate suggests the haversine formula again. Simple demonstration of incorrectness: midpoint from A to B if A and B have same latitude is greater than A and B's latitude (northern hemisphere GC), but this function returns A/B's latitude..

For consistency with other library functions, this would be better added to the CLLocation extensions, with signature something like `func midPoint (to: CLLocation) -> CLLocation` (it's not really fundamental to PolylineNode). And I think these sorts of calculations should have unit test coverage, using a third party source like the calculator at the top of https://www.movable-type.co.uk/scripts/latlong.html, before being added.

However, the way you're using `midPoint()` now won't get into significant errors. So to get you moving, how about a docstring comment that the points must be close together, and an `assert()` to crash and burn if they're not?

Reply:
Moved this method to the CLLocation extension, added docstrings, and renamed it to make it scary (and better describe its limitations).